book/invalidation.md

```diff
@@ -167,6 +167,79 @@ API!
 
 [editing-hates-you]: https://lord.io/text-editing-hates-you-too/
 
+
+Why Invalidation?
+=================
+
+Fundamentally, the reason editing this page is slow in our browser is
+that this page is pretty big.[^other-reasons] After all, it's not
+handling the keypress that's slow---that just appends a character to a
+`Text` node, which takes almost no time. It's what happens afterwards
+that takes time: the browser has to rerender the whole page. Even tiny
+changes take a long time on a large page.
+
+[^other-reasons]: I'm sure there are all sorts of performance
+    improvements possible without implementing the invalidation
+    techniques from this chapter, but they are essential toward
+    achieving incrementalism, which is a kind of asymptotic guarantee
+    that simple optimizations won't achieve.
+
+If we want interactions to be fast, even on large, complex pages,
+we're going to need re-rendering the page to take time *proportional
+to the size of the change*, not proportional to the size of the page.
+This property is sometimes called *incrementality*, and it really
```

Comment:
I don't think I have heard this name, maybe it's a thing in other communities?
Anyway, I think we can coin it and call it the *principle of incremental performance.*

Here would be a good place to explain why this is important:
for developers, so that they can predict the performance
of the web pages they create, and important for users, so they can predict the
performance of interactions (typing: fast; loading a new page: relatively slow).
We can then follow up with commentary about how
the inversion-of-control/constraint programming model of HTML and CSS makes it
hard to achieve this principle, and this chapter is all about how to do so.

Reply:
Great name!